src/helpers.ts

```diff
@@ -0,0 +1,154 @@
+import * as DidKey from '@digitalcredentials/did-method-key';
+import * as DidWeb from '@interop/did-web-resolver';
+import { CryptoLD } from '@digitalcredentials/crypto-ld';
+
+// Crypto library for linked data
+const cryptoLd = new CryptoLD();
+cryptoLd.use(Ed25519VerificationKey2020);
+cryptoLd.use(X25519KeyAgreementKey2020);
+
+// DID drivers
+const didWebDriver = DidWeb.driver({ cryptoLd });
+const didKeyDriver = DidKey.driver();
+
+// Document loader
+const documentLoader = securityLoader().build();
+
+// DID method used to sign credentials
+export enum DidMethod {
+  Key = 'key',
+  Web = 'web'
+}
+
+// Type definition for signCredential method input
+interface SignCredentialOptions {
+  credential: any;
+  didMethod: DidMethod;
+  didSeed: string;
+  didWebUrl?: string;
+}
+
+// Type definition for getSigningKeys method input
+interface GetSigningKeysOptions {
+  didMethod: DidMethod;
+  didSeed: string;
+  didWebUrl?: string;
+}
+
+// Type definition for getSigningKeys method output
+interface GetSigningKeysResult {
+  didDocument: any;
+  issuerDid: string;
+  keyPairs: Map<string, any>;
+  verificationMethod: string;
+}
+
+// signs credential
+export async function signCredential({
+  credential,
+  didMethod,
+  didSeed,
+  didWebUrl
+}: SignCredentialOptions): Promise<VerifiableCredential> {
+  const {
+    keyPairs,
+    verificationMethod
+  } = await getSigningMaterial({
+    didMethod,
+    didSeed,
+    didWebUrl
+  });
+  const key = keyPairs.get(verificationMethod);
+  const date = (new Date()).toISOString();
+  const suite = new Ed25519Signature2020({ key, date });
+  return sign({
+    credential,
+    documentLoader,
+    suite
+  });
+}
+
+// retrieves signing material
+export async function getSigningMaterial({
+  didMethod,
+  didSeed,
+  didWebUrl
+}: GetSigningKeysOptions)
+: Promise<GetSigningKeysResult> {
+  let didDocument;
+  let keyPairs;
+  const didSeedBytes = decodeSeed(didSeed);
+  switch (didMethod) {
+    case DidMethod.Key:
+      ({ didDocument, keyPairs } = await didKeyDriver.generate({
+        seed: didSeedBytes
+      }));
+      break;
+    case DidMethod.Web:
+      ({ didDocument, keyPairs } = await didWebDriver.generate({
+        seed: didSeedBytes,
+        url: didWebUrl
+      }));
+      break;
+    default:
+      throw new Error(
+        '"didMethod" must be one of the following values: ' +
+        `${Object.values(DidMethod).map(v => `'${v}'`).join(', ')}.`
+      );
+  }
+  const issuerDid = didDocument.id;
+  const verificationMethod = extractId(didDocument.assertionMethod[0]);
+  return {
+    didDocument,
+    issuerDid,
+    keyPairs,
+    verificationMethod
+  };
+}
+
+// decodes system data as JSON
+export function decodeSystemData(text: string): any {
+  return JSON.parse(decodeBase64AsAscii(text));
+}
+
+// encodes ASCII text as Bas64
+export function encodeAsciiAsBase64(text: string): string {
+  return Buffer.from(text).toString('base64');
```

Comment:
The `Buffer` usage here reminds me -- might it make sense (in the future) to separate the lightweight client (for checking the status) from the issuing (status allocation etc) code, for use in the browser?